Tests/OnlineRetailPortal.Tests/ProductControllerFixture.cs

```diff
@@ -10,72 +11,115 @@ namespace OnlineRetailPortal.Tests
 {
     public class ProductControllerFixture
     {
-        static IProductStoreFactory storeFactory;
-        private IProductService _productService = new Services.ProductService(storeFactory);
+        private static IProductStoreFactory storeFactory = new ProductStoreFactory();
+        private static IProductService _productService = new Services.ProductService(storeFactory);
+        ProductsController productsController = new ProductsController(_productService);
         [Fact]
         public async void ToGetListOfProducts()
         {
             //Arrenge 
-            int pageNo = 2;
+            int pageNo = 1;
             int pageSize = 1;
-            ProductsController productsController = new ProductsController(_productService);
-            GetProductsServiceRequest getProductsRequest = GetRequest(pageNo, pageSize);
-            GetProductsResponse getProductsServiceResponse = GetExpectedResponse();
 
+            GetProductsResponse expectedResponse = GetExpectedResponse();
 
             //Act
             GetProductsResponse actualResponse = await productsController.GetProductsAsync(pageNo, pageSize);
-            //var createdResult = product as Task<GetProductResponse>;
-
-            //result
-            //Assert.Equal("IPhone", product.Product.Id);
-            Assert.Equal(1, actualResponse.PagingInfo.TotalPages);
-            //Assert.AreEqual(values[1], "value2");
 
+            //Test
+            Assert.Equal(expectedResponse.PagingInfo.PageNumber, actualResponse.PagingInfo.PageNumber);
+            Assert.Equal(expectedResponse.PagingInfo.PageSize, actualResponse.PagingInfo.PageSize);
+            for (var i = 0; i < expectedResponse.Products.Count; i++)
+            {
+                Assert.Equal(expectedResponse.Products[i].Id, actualResponse.Products[i].Id);
+                Assert.Equal(expectedResponse.Products[i].HeroImage, actualResponse.Products[i].HeroImage);
+                Assert.Equal(expectedResponse.Products[i].PostDateTime.ToString(), actualResponse.Products[i].PostDateTime.ToString());
+                Assert.Equal(expectedResponse.Products[i].Name, actualResponse.Products[i].Name);
+                Assert.Equal(expectedResponse.Products[i].Price.IsNegotiable, actualResponse.Products[i].Price.IsNegotiable);
+                Assert.Equal(expectedResponse.Products[i].Price.Money.Amount, actualResponse.Products[i].Price.Money.Amount);
+                Assert.Equal(expectedResponse.Products[i].Price.Money.Currency, actualResponse.Products[i].Price.Money.Currency);
+            }
         }
+        [Fact]
+        public async void To_product_By_Id()
```

Comment:
rename the test case name to something meaningful To_product_By_Id() does not justify what the test case is trying to test

Reply:
done